### PR TITLE
[usb] HID mouse driver.

### PR DESCRIPTION
**Pull request by MichalBlk**
This PR contains changes introduced in #1137.

### CODE REVIEW

**Comment by MichalBlk** [comment]
This PR is once again ready for code review.

File: sys/drv/hidm.c [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    int error = usb_buf_wait(buf);
+    if (!error) {
+      hidm_process_in_report(hidm, &report);
+      hidm->prev_buttons = report.buttons;
+      continue;
+    }
+
+    /* Recover and reenable the transfer. */
+    if ((error = usb_unhalt_endpt(dev, USB_TFR_INTERRUPT, USB_DIR_INPUT)))
+      panic("Unable to unhalt an endpoint");
```
Comment (cahirwpz) [inline sys/drv/hidm.c:133]: This is a bit worrying. Can we react in any reasonable way here?

**Review by cahirwpz — COMMENTED** [review]
Some clarifications are needed. I definitely need to talk to you about HID reports.

File: sys/drv/hidm.c [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+    uint16_t keycode = evdev_hid2btn(i);
+    evdev_push_key(hidm->evdev, keycode, cur);
+    evdev_sync(hidm->evdev);
+  }
+}
+
+static void hidm_process_in_report(hidm_state_t *hidm,
+                                   hidm_in_report_t *report) {
+  /* Hnadle buttons. */
```
[the remark this answers is not in the document]
Comment (MichalBlk) [reply sys/drv/hidm.c]: Fixed.

File: sys/drv/hidm.c [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  hidm_process_buttons(hidm, report->buttons);
+
+  /* Handle the X-axis. */
+  int8_t x = report->x;
+  if (x) {
+    evdev_push_event(hidm->evdev, EV_REL, REL_X, x);
+    evdev_sync(hidm->evdev);
```
[the remark this answers is not in the document]
Comment (MichalBlk) [reply sys/drv/hidm.c]: I checked and it turns out that at most one sync should be sent while processing a single report. Fixed.

File: sys/drv/hidm.c (hunk 1 above)
Comment (MichalBlk) [reply sys/drv/hidm.c:138]: FTTB, I left a command pointing out that we should implement evdev device unregistering.

File: sys/drv/hidm.c [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+  /* Register an interrupt input transfer. */
+  usb_data_transfer(dev, buf, &report, sizeof(hidm_in_report_t),
+                    USB_TFR_INTERRUPT, USB_DIR_INPUT);
+
+  for (;;) {
+    /* Wait for the data to arrvie. */
+    int error = usb_buf_wait(buf);
+    if (!error) {
+      hidm_process_in_report(hidm, &report);
```
[the remark this answers is not in the document]
Comment (MichalBlk) [reply sys/drv/hidm.c]: Done.

**Review by cahirwpz — APPROVED** [review]

**Pull request merged**